cmd/lotus-chainwatch/sync.go

```diff
@@ -146,24 +148,26 @@ func syncHead(ctx context.Context, api api.FullNode, st *storage, ts *types.TipS
 			}
 
 			if len(bh.Parents) == 0 { // genesis case
-				ts, _ := types.NewTipSet([]*types.BlockHeader{bh})
-				aadrs, err := api.StateListActors(ctx, ts.Key())
+				genesisTs, _ := types.NewTipSet([]*types.BlockHeader{bh})
+				aadrs, err := api.StateListActors(ctx, genesisTs.Key())
 				if err != nil {
 					log.Error(err)
 					return
 				}
 
 				parmap.Par(50, aadrs, func(addr address.Address) {
-					act, err := api.StateGetActor(ctx, addr, ts.Key())
+					act, err := api.StateGetActor(ctx, addr, genesisTs.Key())
 					if err != nil {
 						log.Error(err)
 						return
 					}
-					ast, err := api.StateReadState(ctx, act, ts.Key())
+					ast, err := api.StateReadState(ctx, act, genesisTs.Key())
 					if err != nil {
 						log.Error(err)
 						return
 					}
+					// NB(frrist): this json-ified state is written to the db in storeActors().
+					// Writing a json string to the database smells bad, can we do better?
```

Comment:
This column is using the json type - https://www.postgresqltutorial.com/postgresql-json/ - which makes it possible to query actor state directly from postgres queries

Reply:
Thanks for the calcification, will remove the comment block.